File: tools_box/logistics/report/logistics_accident_report/logistics_accident_report.py
# Copyright (c) 2018, masonarmani38@gmail.com and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe

def execute(filters=None):
	data, conditions = [], ""
	columns = [
		"Date:Date:100",
		"Vehicle:Link/Vehicle:100",
		"Type:Data:100",
		"Description:Data:100",
		"Driver:Link/Driver:120",
		"Location:Data:120",
		"Resolution:Data:120",
		"Total Cost:Currency:120"
	]

	if filters.get('from') and filters.get('to'):
		conditions += " and date BETWEEN DATE('{from}') and DATE('{to}')"

	if filters.get('vehicle'):
		conditions += " and vehicle = '{vehicle}'"

	if filters.get('type') :
		conditions += " and driver = '{driver}'"

	data = frappe.db.sql("""select p.date, p.vehicle, p.type, p.description, p.driver,p.location, p.resolution, p.total_cost 
							from `tabLogistics Accident` p where (1=1) {cond} """
						 .format(cond = conditions.format(**filters)))

	return columns,data

File: tools_box/logistics/report/logistics_accident_report/logistics_accident_report.js
// Copyright (c) 2016, masonarmani38@gmail.com and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Logistics Accident Report"] = {
	"filters": [{
			"fieldname":"from",
			"label": __("From Date"),
			"fieldtype": "Date",
			"width": "80",
			"reqd":1
		},
		{
			"fieldname":"to",
			"label": __("To Date"),
			"fieldtype": "Date",
			"default": frappe.datetime.get_today(),
			"reqd":1
		},
		{
			"fieldname":"vehicle",
			"label": __("Vehicle"),
			"fieldtype": "Link",
			"options": "Vehicle",
			"reqd":0
		},
		{
			"fieldname":"driver",
			"label": __("Driver"),
			"fieldtype": "Link",
			"options": "Driver",
			"reqd":0
		},
	]
}
